### ISSUE TITLE
Merge kubekey-k3s with cluster-api-k3s

### ISSUE THREAD

**Issue by nasusoba** [kubesphere/kubekey#2417]
### Describe this feature

Hi, I am from [cluster-api-k3s](https://github.com/k3s-io/cluster-api-k3s). I am seeing that there is a duplicated k3s cluster api, kubekey-k3s. Previously, there was a discussion about merging the two CAPIs [here](https://github.com/kubernetes-sigs/cluster-api/pull/9577), and I would love to spend some time to see if a merge is possible.

The main reason for a merge is to 
* avoid dupliacted feature work or bug fix,
    *  e.g., we have been working on management cluster communication with workload cluster etcd, and I believe this is also useful for kubekey
* have a joint effort to make cluster api k3s more stable!

I have also add [an issue in cluster-api-k3s repo](https://github.com/k3s-io/cluster-api-k3s/issues/142).

### Describe the solution you'd like

Plan in my mind:

1. Understand diff between kubekey-k3s and cluster-api k3s.
   (==> if possible to merge)
2. Contribute the code from kubekey-k3s back to cluster-api k3s.
3. Integrate with kubekey test to verify moving to cluster-api k3s does not break anything.
4. Deprecate kubekey-k3s.

### Additional information

cluster-api-k3s has now supported [airgapped install](https://github.com/k3s-io/cluster-api-k3s/blob/main/samples/docker/air-gapped/README.md). So one diff could be resolved. Could someone help me understand if there is other diff? Thanks so much! : )

**Comment by redscholar** [comment]
Of course, it would be great if you could take it on. We’re happy to contribute to the K3s community.
